### PR TITLE
Reorder functions & structures

### PR DESCRIPTION
**Pull request by NekoSelena**

### CODE REVIEW

**Review by Trompettesib — APPROVED** [review]
👌

**Review by huntears — COMMENTED** [review]
This is fine, but this branch is based of the 1.19.3 branch which is still not merged, so let's wait for it to be merged before merging this one.

**Review by Trompettesib — APPROVED** [review]
blood everywhere on me
merge was so painfull, I jumped out of the window 4 times

File: cubic-server/protocol/ClientPackets.hpp
Comment (STMiki) [inline cubic-server/protocol/ClientPackets.hpp:106]: use `uint8_t` instead of `int32_t`

**Review by STMiki — CHANGES_REQUESTED** [review]
👌

File: cubic-server/protocol/ClientPackets.cpp
Comment (STMiki) [inline cubic-server/protocol/ClientPackets.cpp:83]: remove cast
```suggestion
        animId, addByte);
```

**Review by STMiki — APPROVED** [review]
👌

**Comment by Trompettesib** [comment]
we need to find a way for using correctly enums without having to cast

**Review by Trompettesib — APPROVED** [review]
👌

File: cubic-server/protocol/ClientPackets.cpp
Comment (Trompettesib) [reply cubic-server/protocol/ClientPackets.cpp]: We can't

Comment (Trompettesib) [reply cubic-server/protocol/ClientPackets.cpp]: we need to cast somewhere

File: cubic-server/protocol/ClientPackets.cpp
Comment (STMiki) [inline cubic-server/protocol/ClientPackets.cpp:84]: Same `uint8_t` 🤡

**Review by STMiki — CHANGES_REQUESTED** [review]

File: cubic-server/protocol/ClientPackets.cpp
Comment (Trompettesib) [reply cubic-server/protocol/ClientPackets.cpp:84]: But I can't remove the cast. If I do so, because animId is a EntityAnimationID, and not a uint8_t it will not compile

Comment (Trompettesib) [reply cubic-server/protocol/ClientPackets.cpp:84]: We have the same problem everywhere else

Comment (STMiki) [reply cubic-server/protocol/ClientPackets.cpp:84]: Oh yeah, enum class things 😅

**Review by STMiki — APPROVED** [review]

**Pull request merged**
